### PR TITLE
[TS] Updated tsconfigs design for generated app

### PR DESCRIPTION
**Pull request by Convly**
### In this PR
- [x] Replace the `tsconfig-admin.json` and `tsconfig-server.json` files by `./tsconfig.json` and `./src/admin/tsconfig.json`
- [x] Adds a dedicated `tsconfig.json` for each local plugins (./src/plugins/**/tsconfig.json) + scripts to handle develop, build & start.

### Why is it needed?

See #12927 

### How to test it?

- clone the monorepository
- create a new Strapi project using the cli (in the monorepo) with --ts
- the project should have been created with 2 tsconfig.json files (one at the root level & another one in src/admin)
- try to yarn develop inside this project
- everything should work correctly

### Related issue(s)/PR(s)

#12927

### CODE REVIEW

File: packages/core/admin/webpack.config.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ module.exports = ({
     const tsChecker = new ForkTsCheckerPlugin({
       typescript: {
         // FIXME
-        configFile: path.join(appDir, 'tsconfig-admin.json'),
+        configFile: path.join(appDir, 'src', 'admin', 'tsconfig.json'),
```
Comment (soupette) [inline packages/core/admin/webpack.config.js:57]: You can remove the FIXME comment.

File: packages/core/strapi/lib/commands/build.js [hunk 2]
```diff
@@ -24,7 +24,8 @@ module.exports = async ({ optimization, forceBuild = true }) => {
   await buildAdmin({
     buildDestDir,
     forceBuild,
-    isTSProject,
+    // TODO: Delegate the is-ts-project check to the admin
```
Comment (soupette) [inline packages/core/strapi/lib/commands/build.js:27]: What's the TODO for?

Comment (soupette) [reply packages/core/strapi/lib/commands/build.js:27]: You can remove that part and just check if we have a tsconfig.json file in the src/admin folder instead.

File: packages/core/strapi/lib/commands/develop.js [hunk 3]
```diff
@@ -61,6 +61,7 @@ const primaryProcess = async ({ distDir, appDir, build, isTSProject, watchAdmin,
       await buildAdmin({
         buildDestDir: distDir,
         forceBuild: false,
+        // TODO: delegates the is-ts-project check to the admin
```
Comment (soupette) [inline packages/core/strapi/lib/commands/develop.js:64]: Same here.

File: packages/core/strapi/lib/commands/build.js (hunk 2 above)
Comment (soupette) [reply packages/core/strapi/lib/commands/build.js:27]: But you'll have to know where to put the build folder however it's located inside the dist one only if the backend has a tsconfig.json file.

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/core/strapi/lib/commands/build.js (hunk 2 above)
Comment (Convly) [reply packages/core/strapi/lib/commands/build.js:27]: Discussed & fixed in your branch/side, marking as closed & removing the todo

File: packages/core/strapi/lib/commands/develop.js (hunk 3 above)
Comment (Convly) [reply packages/core/strapi/lib/commands/develop.js:64]: Same response as above

**Review by soupette — APPROVED** [review]

**Pull request merged**